Tests/CoreDataFeedStoreTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  CoreDataFeedStoreTests.swift
+//  Tests
+//
+//  Created by Shilpa Bansal on 23/02/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import XCTest
+@testable import FeedStoreChallenge
+
+class CoreDataFeedStoreTests: XCTestCase, FeedStoreSpecs {
+	override func setUp() {
+		super.setUp()
+	}
+	
+	override func tearDown() {
+		super.tearDown()
+	}
+	
+	func test_retrieve_deliversEmptyOnEmptyCache() throws {
+		let sut = try makeSUT()
+
+		assertThatRetrieveDeliversEmptyOnEmptyCache(on: sut)
+	}
+	
+	func test_retrieve_hasNoSideEffectsOnEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatRetrieveHasNoSideEffectsOnEmptyCache(on: sut)
+	}
+	
+	func test_retrieve_deliversFoundValuesOnNonEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatRetrieveDeliversFoundValuesOnNonEmptyCache(on: sut)
+	}
+	
+	func test_retrieve_hasNoSideEffectsOnNonEmptyCache() throws {
+		let sut = try makeSUT()
+
+		assertThatRetrieveHasNoSideEffectsOnNonEmptyCache(on: sut)
+	}
+	
+	func test_insert_deliversNoErrorOnEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatInsertDeliversNoErrorOnEmptyCache(on: sut)
+	}
+	
+	func test_insert_deliversNoErrorOnNonEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatInsertDeliversNoErrorOnNonEmptyCache(on: sut)
+	}
+	
+	func test_insert_overridesPreviouslyInsertedCacheValues() throws {
+		let sut = try makeSUT()
+		assertThatInsertOverridesPreviouslyInsertedCacheValues(on: sut)
+	}
+	
+	func test_delete_deliversNoErrorOnEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatInsertDeliversNoErrorOnEmptyCache(on: sut)
+	}
+	
+	func test_delete_hasNoSideEffectsOnEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatDeleteHasNoSideEffectsOnEmptyCache(on: sut)
+	}
+	
+	func test_delete_deliversNoErrorOnNonEmptyCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatDeleteDeliversNoErrorOnNonEmptyCache(on: sut)
+	}
+	
+	func test_delete_emptiesPreviouslyInsertedCache() throws {
+		let sut = try makeSUT()
+		
+		assertThatDeleteEmptiesPreviouslyInsertedCache(on: sut)
+	}
+	
+	func test_storeSideEffects_runSerially() throws {
+		let sut = try makeSUT()
+		
+		assertThatSideEffectsRunSerially(on: sut)
+	}
+	
+	private func testSpecificURL() -> URL? {
```

Comment:
This method never retuns `nil`, so the return type shouldn't be `optional`.

```suggestion
	private func testSpecificURL() -> URL {
```